### PR TITLE
Replace `jsdom` with `browser-env` in browser recipe

### PR DESCRIPTION
**Pull request by lukechilds**
Fixes https://github.com/avajs/ava/issues/882

### CODE REVIEW

File: docs/recipes/browser-testing.md [hunk 1]
```diff
@@ -7,29 +7,33 @@ An example of this is React, at least if you want to use ReactDOM.render and sim
...
 
-Install [jsdom](https://github.com/tmpvar/jsdom).
+Install [node-browser-environment](https://github.com/lukechilds/node-browser-environment).
 
-> A JavaScript implementation of the WHATWG DOM and HTML standards, for use with node.js
+> Simulates a global browser environment using jsdom.
 
 ```
-$ npm install --save-dev jsdom
+$ npm install --save-dev node-browser-environment
```
Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: That's a very verbose module name. Why not simply `browser-env`? Node is already implied.

File: docs/recipes/browser-testing.md [hunk 2]
```diff
@@ -7,29 +7,33 @@ An example of this is React, at least if you want to use ReactDOM.render and sim
...
 
 Create a helper file and place it in the `test/helpers` folder. This ensures AVA does not treat it as a test.
 
 `test/helpers/setup-browser-env.js`:
 
 ```js
-global.document = require('jsdom').jsdom('<body></body>');
-global.window = document.defaultView;
-global.navigator = window.navigator;
+require('node-browser-environment')();
```
Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: Should use `import`, not `require`.

File: docs/recipes/browser-testing.md [hunk 3]
```diff
@@ -7,29 +7,33 @@ An example of this is React, at least if you want to use ReactDOM.render and sim
...
 ```
 
-## Configure tests to use jsdom
+By default `node-browser-environment` will add all global browser variables to node's global scope creating a full browser environment. This should have good compatibility with most front end libraries, however it's generally not a good idea to create loads of global variables if you don't need to. If you know exactly which browser variables you need you can pass an array of required properties.
+
+```js
+require('node-browser-environment')(['window', 'document', 'navigator']);
```
Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: Use `import`

File: docs/recipes/browser-testing.md [hunk 4]
```diff
@@ -7,29 +7,33 @@ An example of this is React, at least if you want to use ReactDOM.render and sim
...
 
 ```js
-global.document = require('jsdom').jsdom('<body></body>');
-global.window = document.defaultView;
-global.navigator = window.navigator;
+require('node-browser-environment')();
 ```
 
-## Configure tests to use jsdom
+By default `node-browser-environment` will add all global browser variables to node's global scope creating a full browser environment. This should have good compatibility with most front end libraries, however it's generally not a good idea to create loads of global variables if you don't need to. If you know exactly which browser variables you need you can pass an array of required properties.
```
Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: node => Node.js

front end => front-end

loads of => lots of

**Comment by sindresorhus** [comment]
@lukechilds Are you committed to keeping this module up to date with `jsdom`?

**Comment by lukechilds** [comment]
@sindresorhus Yep, I've pinned the version number for `jsdom` to stop any `jsdom` breaking changes breaking AVA tests and I've got Greenkeeper set up to notify me of `jsdom` updates. If you look through the commit history you'll see I normally update this within 24 hours.

I've got no intention of abandoning this module but I'm happy to add you or anyone else on the AVA team as a maintainer if you want.

Working on your requested changes now 👍

**Comment by lukechilds** [comment]
@sindresorhus Done requested changes 🙂

File: docs/recipes/browser-testing.md (hunk 1 above)
Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: It's pretty easy. You just bump major on the existing one and deprecate with a message saying it's been renamed. We bump major first so the deprecation notice doesn't annoy existing users until they bump a version. I would personally do it as that module is going to be used a lot and `browser-environment` would look nicer in the import statement, but your choice ;)

Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: Actually, https://github.com/CamShaft/browser-env is deprecated, so you could just request the module name.

https://docs.npmjs.com/misc/disputes

Comment (lukechilds) [inline docs/recipes/browser-testing.md]: Good spot on the deprecation, yeah I'll rename it. If you wanna hold off on merging this for now I'll update you when it's sorted 👍

Quick Q: Why bump major first? Wouldn't it be better to notify existing users? If they've installed with the default `^` range they'll be expecting to get SemVer minor/patch updates automatically.

Comment (sindresorhus) [inline docs/recipes/browser-testing.md]: @lukechilds Not very relevant in this situation, as it's a top-level module, so maybe not necessary, but when your module is deep in a dependency tree, people will start opening issues on every module that depends on it somewhere in the dependency tree about updating, which is very annoying for the maintainers and a lot of wasted time. Users think warnings are critical.

Comment (lukechilds) [inline docs/recipes/browser-testing.md]: @sindresorhus `node-browser-environment` is dead, long live `browser-env`! 🎉

**Comment by lukechilds** [comment]
I'm thinking something along the lines of:

`test/my.dom.test.js`:

```js
import test from 'ava';

test('Insert to DOM', t => {
  const div = document.createElement('div');
  document.body.appendChild(div);

  t.is(document.querySelector('div'), div);
});
```

It could be as well as the current React test or instead of. What do you think?

**Comment by sindresorhus** [comment]
Good idea. Should be in addition to the React one.

**Comment by lukechilds** [comment]
@sindresorhus Done 👍

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**